### DISCUSSION TITLE
For autocomplete, any way to go back to normal mode after autocomplete selection is inserted?

### DISCUSSION THREAD

**Discussion by jamischarles** [neoclide/coc.nvim#3816] [Q&A]
LOVE coc!

This is more of a question.

**I'm using autocomplete and this keymapping:**
`inoremap <expr> <cr> pumvisible() ? "\<C-y>" : "\<C-g>u\<CR>"`

Works great. 

However, **I'm trying to automatically go to `normal` mode after the insertion is completed**.

For the life of me I cannot figure out how to do that.
`inoremap <expr> <cr> pumvisible() ? "\<C-y>\<Esc>" : "\<C-g>u\<CR>"` Ignores the completion, and goes to normal mode
`inoremap <expr> <cr> pumvisible() ? "\<C-y>\<C-]>" : "\<C-g>u\<CR>"` Ignores the completion, and goes to normal mode
`inoremap <expr> <cr> pumvisible() ? "\<C-y>Hello" : "\<C-g>u\<CR>"`  Ignores completion, inserts `Hello`.

Any ideas?

Thanks and keep up the great work!

**Comment by chemzqm** [comment]
You have to write a function with timer to feed `<esc>`
It will leave select mode of snippet which should be bad
